pkg/verify/verify.go

```diff
@@ -283,9 +223,54 @@ func (v *releaseVerifier) Verify(ctx context.Context, releaseDigest string) erro
 		}
 		return fmt.Errorf("unable to locate a valid signature for one or more sources")
 	}
+
+	v.cacheVerification(releaseDigest, signedWith)
+
 	return nil
 }
 
+// Signatures returns a copy of any cached signatures that have been validated
+// so far. It may return no signatures.
+func (v *ReleaseVerifier) Signatures() map[string][][]byte {
```

Comment:
Why would you want all of these?  I think `Verify` is a sufficient API.  If this is "which sigs will `oc` have to pack into a ConfigMap for disconnected folks?", I'd rather have:

```go
 func (v *ReleaseVerifier) Signatures(digest string) [][]byte
```

or some such.

Reply:
I want to get all of them in bulk at once for saving.

Reply:
> I want to get all of them in bulk at once for saving.

How about configuring a helper that gets called on signatures when they are added to the cache?  And another that is called when they are evicted?  Then your mirror case or "protect from the sig store going down" case could be addressed by attaching a collecter to the added-to-cache callback.  And you'd have a way to evict ConfigMaps in the "protect from the sig store going down" case when we dropped them from the in-memory cache.

Reply:
a controller feels super heavy weight.  Why do we need something more complex than "get" and "write"?

Reply:
> Why do we need something more complex than "get" and "write"?

Maybe those are fine.  But I like decoupled logic so the verifier can say "I verify sigs, and if you want to bolt a cache or whatever on, here's where you hand over your callbacks".  Then the code that handles verification is all nice and compact, and the code that handles caching can live somewhere else all nice and compact, and the package that wants caching verification can glue the two together to get what they want.